app/resources/services/integrations/coil.js

```diff
@@ -0,0 +1,167 @@
+    const user = async () => {
+      await User.findByPk(id)
+    }
+    done(null, user)
+  })
+}
+
+// Only initialize Patreon auth strategy if the env vars are set.
+if (process.env.COIL_CLIENT_ID && process.env.COIL_CLIENT_SECRET) {
+  initCoil()
+}
+
+exports.get = (req, res, next) => {
+  if (!process.env.COIL_CLIENT_ID || !process.env.COIL_CLIENT_SECRET) {
+    res.status(500).json({ status: 500, msg: 'Coil integration unavailable.' })
+  }
+
+  /*
+    at this point,the request has user info from auth0 that we passed to this route
+    auth0 nickname == user.id in our internal db that we'll need later for lookup
+
+    this might not be available if the user isn't signed in
+      we could add error handiling for this, but also they _should_ only ever be
+      getting here from a button that you only see when you're signed in..
+    */
+  passport.authorize('coil', {
+    state: req.user.sub,
+    failureRedirect: '/error'
+  })(req, res, next)
+}
+
+exports.callback = (req, res, next) => {
+  if (!process.env.COIL_CLIENT_ID || !process.env.COIL_CLIENT_SECRET) {
+    res.status(500).json({ status: 500, msg: 'Coil integration unavailable.' })
+  }
+  passport._strategies.coil._oauth2.setAuthMethod('BASIC')
+  passport.authorize('coil', {
+    failureRedirect: '/error'
+  })(req, res, next)
+}
+
+const getBTPToken = async (accessToken) => {
+  try {
+    const requestConfig = {
+      method: 'post',
+      url: 'https://api.coil.com/user/btp',
+      headers: {
+        Authorization: 'Bearer ' + accessToken,
+        content_type: 'application/x-www-form-urlencoded'
+      }
+    }
+    const response = await axios(requestConfig)
+    const token = response.data.btpToken
+    return token
+  } catch (error) {
+    logger.error(error)
+  }
+}
+
+/**
+ * connects the third party profile with the database user record
+ * pass third party profile data here, construct an object to save to user DB
+ */
+exports.connectUser = async (req, res, next) => {
+  // in passport, using 'authorize' attaches user data to 'account'
+  // instead of overriding the user session data
+  const databaseUser = req.account
+  const identity = {
+    provider: req.profile.provider,
+    user_id: req.profile.id,
+    access_token: req.profile.access_token,
+    refresh_token: req.profile.refresh_token
+  }
+
+  try {
+    // TODO: put this in the template somehow, for now in request session (or maybe cookie?)
+    // TODO this returns a 403 (maybe 404, thats what I'm getting on the router now)
+    const btpToken = await getBTPToken(req.profile.access_token)
+    req.session.btpToken = btpToken
+    await User.update(
+      {
+        identities: [identity]
+      },
+      { where: { auth0Id: databaseUser.auth0Id }, returning: true }
+    )
+    // todo: should we call next() here instead of redirect to root?
+    next()
+  } catch (error) {
+    // what would we want to do here?
+    logger.error(error)
+    res.redirect('/error')
```

Comment:
As we discussed, we'd want to handle errors and surface them in the UI as dismissable items, rather than redirect to an error page, which breaks the user experience.

Reply:
since this is more of a frontend feature, I'd like to think of that as out of scope for this PR. I'm expecting it'll be easier to handle in a unified way once we have this and the patreon dialog merged in.